presto-hive/src/main/java/com/facebook/presto/hive/HiveBucketing.java

```diff
@@ -279,10 +282,13 @@ private static int hashBytes(int initialValue, Slice bytes)
         return result;
     }
 
-    public static Optional<HiveBucketHandle> getHiveBucketHandle(Table table)
+    public static Optional<HiveBucketHandle> getHiveBucketHandle(ConnectorSession session, Table table)
     {
         Optional<HiveBucketProperty> hiveBucketProperty = table.getStorage().getBucketProperty();
         if (!hiveBucketProperty.isPresent()) {
+            if (table.getTableType().equals(TEMPORARY_TABLE)) {
+                return Optional.of(HiveBucketHandle.createVirtualBucketHandle(getCteVirtualBucketCount(session)));
```

Comment:
Main fix, in case of unpartitioned temp tables, we now switch to virtual bucketing.